index.test-d.ts

```diff
@@ -12,3 +12,4 @@ expectType<([string | number, string | number])[]>(
 	arrify(new Map<string | number, string | number>([[1, 2], ['a', 'b']]))
 );
 expectType<number[]>(arrify(new Set([1, 2])));
+expectType<ReadonlyArray<number>>(arrify([Number()] as const));
```

Comment:
This assertion currently doesn't do what you expect. Currently (until https://github.com/SamVerschueren/tsd/pull/21 is merged) this is an assignability check. A mutable array is always assignable to an immutable array parameter/binding. This means that this check is a noop if you're trying to verify that the result of `arrify` is a readonly array:
```ts
const arr: string[] = [];
expectType<readonly string[]>(arr);
```
The code above will pass without triggering an assertion error.

To workaround this issue please refactor this to:
```suggestion
expectError(arrify(['🦄'] as const).push(''));
```